src/app/plugins/ariana.plugin.ts

```diff
@@ -0,0 +1,79 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, IHttpResponse } from '../../common/types';
+
+export class ArianaPlugin extends Plugin {
+  public name: string = 'Ariana Plugin';
+  public description: string = 'Get a random picture of Ariana Grande.';
+  public usage: string = 'ariana';
+  public pluginAlias = ['ari'];
+  public permission: ChannelType = ChannelType.Bot;
+
+  private _API_URL: string = 'https://www.gettyimages.com/photos/ariana-grande?page=';
+  private reImgs: RegExp = new RegExp(/<img class="gallery-asset__thumb gallery-mosaic-asset__thumb"(.*?)>/g);
+  private reThumb: RegExp = new RegExp(/https:\/\/(.*?)"/g);
+  private rePages: RegExp = new RegExp(/<span class="PaginationRow-module__lastPage___2pChH">(.*?)<\/span>/g);
+  private rePage: RegExp = new RegExp(/>(.*?)</g);
+  private nPages: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+
+    // get total number of pages of Ariana
+    this.container.httpService
+      .get(this._API_URL + '1')
+      .then((res: IHttpResponse) => {
+        const data: string = res.data;
+        const pagesSearch: RegExpMatchArray | null = data.match(this.rePages);
+
+        if (!pagesSearch) {
+          this.container.loggerService.warn('Failed to get Ariana page in constructor.');
+          return;
+        }
+        const pageSearch: RegExpMatchArray | null = pagesSearch[0].match(this.rePage);
```

Comment:
use type Maybe from types.ts for lines 32, 36, and 46.
https://github.com/cs-discord-at-ucf/lion/blob/master/src/common/types.ts < line 176